crates/cext/src/circuit.rs

```diff
@@ -851,6 +852,44 @@ pub struct CInstruction {
     /// The number of parameters for this instruction.
     num_params: u32,
 }
+impl CInstruction {
+    /// Create a `CInstruction` that owns pointers to copies of the information in the given
+    /// `PackedInstruction`.
+    ///
+    /// This must be cleared by a call to `qk_circuit_instruction_clear` to avoid leaking its
+    /// allocations.
+    ///
+    /// Panics if the operation name contains a nul, or if the instruction has non-float parameters.
```

Comment:
Is this meant to be nul or is it a typo? It shows up later in the code so I'm assuming it's some meaning I don't know of, but in case it isn't.

```suggestion
    /// Panics if the operation name contains a null, or if the instruction has non-float parameters.
```

Reply:
The ASCII `\0` byte is referred to by both "nul" and "null" - the former comes from its three-letter short representation.  We use `nul` elsewhere in the docs for it at the moment, so we might want to tidy up uniformly later.